bundles/alpha.codegen/src/alpha/codegen/writeC/SystemConverter.xtend

```diff
@@ -0,0 +1,472 @@
+		// However, if we need compatibility with the older AlphaZ system,
+		// change the level of indirection to match the number of indices
+		// in the variable's domain.
+		val dataType = Common.alphaVariableType
+		if (oldAlphaZCompatible) {
+			dataType.indirectionLevel = variable.domain.nbIndices
+		}
+		return builder.prepareEntryArgument(variable.name, dataType)
+	}
+	
+	/** Adds a parameter to the function and the assignment statement needed to copy it to the global variable. */
+	def protected static prepareEntryArgument(FunctionBuilder builder, String name, DataType type) {
+		// We use a prefix to differentiate the entry point's argument name
+		// from the global variable it represents.
+		val argName = "_local_" + name
+		
+		// Add the argument as a new parameter for the function.
+		builder.addParameter(type, argName)
+		
+		// Copy the value to the global variable.
+		val copyStmt = Factory.assignmentStmt(name, argName)
+		builder.addStatement(copyStmt)
+	}
+	
+	/** Adds the "if" statements needed to check that the parameters are valid. */
+	def protected checkParameters(FunctionBuilder builder) {
+		val isValid = ConditionalConverter.convert(system.parameterDomain)
+		val parameterCheck = IfStmtBuilder
+			.start(Factory.unaryExpr(UnaryOperator.NOT, isValid))
+			.addStatement(
+				Factory.callStmt("printf", "\"The value of the parameters are invalid.\\n\""),
+				Factory.callStmt("exit", "-1")
+			)
+			.instance
+
+		builder.addComment("Check parameter validity.")
+			.addStatement(parameterCheck)
+			.addEmptyLine
+	}
+	
+	
+	/////////////////////////////////////////////////////////////////////////////
+	// Memory Allocation and Deallocation
+	/////////////////////////////////////////////////////////////////////////////
+	
+	/** Adds the statements needed to allocate memory for the Alpha variables and any necessary flags. */
+	def protected allocateMemory(FunctionBuilder builder) {
+		// Allocate the Alpha local variables first.
+		builder.addComment("Allocate memory for local storage.")
+		system.locals.forEach[builder.allocateMemory(it, false)]
+		builder.addEmptyLine
+		
+		// Allocate memory for, and initialize, all the flags variables.
+		builder.addComment("Allocate and initialize flag variables.")
+		system.outputs.forEach[builder.allocateMemory(it, true)]
+		system.locals.forEach[builder.allocateMemory(it, true)]
+		builder.addEmptyLine
+	}
+	
+	/**
+	 * Adds the calls to "malloc", "mallocCheck", and "memset" as needed
+	 * for an Alpha variable or its flag variable.
+	 */
+	def protected allocateMemory(FunctionBuilder builder, Variable variable, boolean allocateFlag) {
+		// Determine the name and data type of the C variable based on
+		// whether we're allocating memory for the standard variable or the flags variable.
+		val name = allocateFlag ? variable.flagName : variable.name
+		val dataType = allocateFlag ? Common.flagVariableType : Common.alphaValueType
```

Comment:
I think the else branch of this should be `Common.alphaVariableType` instead of `Common.alphaValueType`.

Reply:
Yes, fixed in commit 8eb2cc6